src/libostree/ostree-core.c

```diff
@@ -839,12 +839,13 @@ ostree_checksum_file (GFile            *f,
  * @stbuf (allow-none): Optional stat buffer
  * @objtype: Object type
  * @flags: Flags
- * @out_csum: (out) (array fixed-size=32): Return location for binary checksum
+ * @checksum_buf (out caller-allocates): Buffer to hold SHA256 hex checksum
```

Comment:
Although, sorry to make you rev this again...this might not be introspectable.  I think we'd still need the `(array fixed-size=65)` at least.

So...eh, I guess let's just malloc here and basically use the out-buffer approach for more internal APIs - basically where it matters more is anything we do in bulk, so it's obviously fine to malloc a commit checksum, but we should try to reduce internal allocs around writing files.

Reply:
Updated below! ⬇️